framework/src/dirackernels/VectorPostprocessorPointSource.C

```diff
@@ -20,41 +20,86 @@ VectorPostprocessorPointSource::validParams()
+      "x_coord_name",
+      "x",
+      "<reporter>/<x_coord_name> containing x coordinates.  This is just "
+      "the x_coord_name if providing input for a vector_postprocessor name.");
+  params.addParam<std::string>(
+      "y_coord_name",
+      "y",
+      "<reporter>/<y_coord_name> containing y coordinates.  This is just "
+      "the y_coord_name if providing input for a vector_postprocessor name.");
+  params.addParam<std::string>(
+      "z_coord_name",
+      "z",
+      "<reporter>/<z_coord_name> containing z coordinates.  This is just "
+      "the z_coord_name if providing input for a vector_postprocessor name.");
+  params.addParam<std::string>("value_name",
+                               "value",
+                               "<reporter>/<value_name> containing values.  This is just the "
+                               "value_name if providing input for a vector_postprocessor name.");
+
   return params;
 }
 
 VectorPostprocessorPointSource::VectorPostprocessorPointSource(const InputParameters & parameters)
   : DiracKernel(parameters),
-    _vpp_values(getVectorPostprocessorValue(
-        "vector_postprocessor", getParam<std::string>("value_name"), true)),
-    _x_coord(getVectorPostprocessorValue(
-        "vector_postprocessor", getParam<std::string>("x_coord_name"), true)),
-    _y_coord(getVectorPostprocessorValue(
-        "vector_postprocessor", getParam<std::string>("y_coord_name"), true)),
-    _z_coord(getVectorPostprocessorValue(
-        "vector_postprocessor", getParam<std::string>("z_coord_name"), true))
+    ReporterInterface(this),
+    _vpp_values(
+        isParamValid("vector_postprocessor")
+            ? getReporterValueByName<std::vector<Real>>(
+                  getParam<std::string>("vector_postprocessor") + "/" +
+                  getParam<std::string>("value_name"))
+            : getReporterValueByName<std::vector<Real>>(getParam<std::string>("value_name"))),
+    _x_coord(
+        isParamValid("vector_postprocessor")
+            ? getReporterValueByName<std::vector<Real>>(
+                  getParam<std::string>("vector_postprocessor") + "/" +
+                  getParam<std::string>("x_coord_name"))
+            : getReporterValueByName<std::vector<Real>>(getParam<std::string>("x_coord_name"))),
+    _y_coord(
+        isParamValid("vector_postprocessor")
+            ? getReporterValueByName<std::vector<Real>>(
+                  getParam<std::string>("vector_postprocessor") + "/" +
+                  getParam<std::string>("y_coord_name"))
+            : getReporterValueByName<std::vector<Real>>(getParam<std::string>("y_coord_name"))),
+    _z_coord(isParamValid("vector_postprocessor")
+                 ? getReporterValueByName<std::vector<Real>>(
+                       getParam<std::string>("vector_postprocessor") + "/" +
+                       getParam<std::string>("z_coord_name"))
+                 : getReporterValueByName<std::vector<Real>>(getParam<std::string>("z_coord_name")))
```

Comment:
I don't really like all this duplicate code. I would recommend just having a helper function like this:

```cpp
const std::vector<Real> &
VectorPostprocessorPointSource::getPointDataHelper(const std::string & param)
{
  ReporterName rname = isParamValid("vector_postprocessor") ?
                       ReporterName(getParam<std::string>("vector_postprocessor"), getParam<std::string>(param)) :
                       ReporterName(getParam<std::string>(param));
  return getReporterValueByName<std::vector<Real>>(rname);
}
```

The you can just do this in the constructor:

```cpp
...
_vpp_values(getPointDataHelper("value_name")),
_x_coord(getPointDataHelper("x_coord_name")),
_y_coord(getPointDataHelper("y_coord_name")),
_z_coord(getPointDataHelper("z_coord_name"))
```

Reply:
Also, as an extra precaution, it might be good to make sure we are receiving all the values on all processors:
```cpp
getReporterValueByName<std::vector<Real>>(rname, REPORTER_MODE_REPLICATED)
```

Reply:
Thanks @zachmprince.   I pasted in your changes.